### PR TITLE
Internal renames and cleanup in resource importer scene

### PR DESCRIPTION
**Pull request by aaronfranke**
While I was working on #72753, the cleanups I was doing was making it hard to track the actual behavior changes, so this PR splits off the cleanup work for easy reviewability. This PR does NOT change behavior.

* Rename `_apply_basis_to_scalable_node_collection` to `_apply_scale_to_scalable_node_collection`
* Rename `_apply_permanent_rotation_scale_to_node` to `_apply_permanent_scale_to_descendants`
* Rename the `ScalableNodeCollection` parameter `p_dictionary` to `p_collection`
* Only construct `Vector3(root_scale, root_scale, root_scale)` once.
* Pass the scale to `_apply_permanent_scale_to_descendants` instead of re-calculating it later.
* When applying the root scale, don't needlessly scale and then un-scale the root node.
* In `_apply_scale_to_scalable_node_collection`, remove a useless `if (node_3d) {` check (it's always true).

### CODE REVIEW

**Review by fire — APPROVED** [review]
The review looks good, but need to test on various skeletons that are scaled 100x.

See also our tests from godot tests.

https://github.com/godotengine/godot-tests/tree/master/tests/gltf_skeleton_tests

TL;DR needs to confirm that behaviour didn't change.
